typescript/cli/src/config/ism.ts

```diff
@@ -87,116 +90,197 @@ export async function createIsmConfig(
+    return {
+      type: IsmType.TRUSTED_RELAYER,
+      relayer,
+    };
+  },
+  IsmType.TRUSTED_RELAYER,
+);
 
-export async function createRoutingConfig(
-  context: CommandContext,
-  type: IsmType.ROUTING | IsmType.FALLBACK_ROUTING,
-): Promise<IsmConfig> {
-  const owner = await input({
-    message: 'Enter owner address for routing ISM',
-  });
-  const ownerAddress = owner;
-
-  const chains = await runMultiChainSelectionStep(
-    context.chainMetadata,
-    'Select chains to configure ISM for',
-    true,
-  );
-
-  const domainsMap: ChainMap<IsmConfig> = {};
-  for (const chain of chains) {
-    await confirm({
-      message: `You are about to configure ISM from source chain ${chain}. Continue?`,
+export const createAggregationConfig = callWithConfigCreationLogs(
+  async (context: CommandContext): Promise<AggregationIsmConfig> => {
+    const isms = parseInt(
+      await input({
+        message: 'Enter the number of ISMs to aggregate (number)',
+      }),
+      10,
+    );
+
+    const threshold = parseInt(
+      await input({
+        message: 'Enter the threshold of ISMs for verification (number)',
+      }),
+      10,
+    );
+
+    const modules: Array<IsmConfig> = [];
+    for (let i = 0; i < isms; i++) {
+      modules.push(await createAdvancedIsmConfig(context));
+    }
+    return {
+      type: IsmType.AGGREGATION,
+      modules,
+      threshold,
+    };
+  },
+  IsmType.AGGREGATION,
+);
+
+export const createRoutingConfig = callWithConfigCreationLogs(
+  async (context: CommandContext): Promise<IsmConfig> => {
+    const owner = await input({
+      message: 'Enter owner address for routing ISM',
     });
-    const config = await createIsmConfig(context);
-    domainsMap[chain] = config;
+    const ownerAddress = owner;
+    const requireMultiple = true;
+    const chains = await runMultiChainSelectionStep(
+      context.chainMetadata,
+      'Select chains to configure routing ISM for',
+      requireMultiple,
+    );
+
+    const domainsMap: ChainMap<IsmConfig> = {};
+    for (const chain of chains) {
+      await confirm({
+        message: `You are about to configure routing ISM from source chain ${chain}. Continue?`,
+      });
+      const config = await createAdvancedIsmConfig(context);
+      domainsMap[chain] = config;
+    }
+    return {
+      type: IsmType.ROUTING,
+      owner: ownerAddress,
+      domains: domainsMap,
+    };
+  },
+  IsmType.ROUTING,
+);
+
+export const createFallbackRoutingConfig = callWithConfigCreationLogs(
+  async (context: CommandContext): Promise<IsmConfig> => {
+    const owner = await input({
+      message: 'Enter owner address for fallback routing ISM',
+    });
+    const ownerAddress = owner;
+
+    const chains = await runMultiChainSelectionStep(
+      context.chainMetadata,
+      'Select chains to configure fallback routing ISM for',
+      true,
+    );
+
+    const domainsMap: ChainMap<IsmConfig> = {};
+    for (const chain of chains) {
+      await confirm({
```

Comment:
same here. confirm (y/n) is not used for anything

Reply:
replaced with log